tests/langchain/test_langchain_model_export.py

```diff
@@ -302,6 +322,183 @@ def test_langchain_native_log_and_load_qa_with_sources_chain():
     assert model == loaded_model
 
 
+def load_retriever(persist_directory):
+    embeddings = FakeEmbeddings(size=5)
+    vectorstore = FAISS.load_local(persist_directory, embeddings)
+    return vectorstore.as_retriever()
+
+
+@pytest.mark.skipif(
+    version.parse(langchain.__version__) < version.parse("0.0.194"),
+    reason="Saving RetrievalQA chains requires langchain>=0.0.194",
+)
+def test_log_and_load_retrieval_qa_chain(tmp_path):
+    # Create the vector db, persist the db to a local fs folder
+    loader = TextLoader("tests/langchain/state_of_the_union.txt")
+    documents = loader.load()
+    text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=0)
+    docs = text_splitter.split_documents(documents)
+    embeddings = FakeEmbeddings(size=5)
+    db = FAISS.from_documents(docs, embeddings)
+    persist_dir = str(tmp_path / "faiss_index")
+    db.save_local(persist_dir)
+
+    # Create the RetrievalQA chain
+    retrievalQA = RetrievalQA.from_llm(llm=OpenAI(), retriever=db.as_retriever())
+
+    # Log the RetrievalQA chain
+    with mlflow.start_run():
+        logged_model = mlflow.langchain.log_model(
+            retrievalQA,
+            "retrieval_qa_chain",
+            loader_fn=load_retriever,
+            persist_dir=persist_dir,
+        )
+
+    # Remove the persist_dir
+    shutil.rmtree(persist_dir)
+
+    # Load the chain
+    loaded_model = mlflow.langchain.load_model(logged_model.model_uri)
+    assert loaded_model == retrievalQA
+
+    # Serve the chain
+    loaded_pyfunc_model = mlflow.pyfunc.load_model(logged_model.model_uri)
+    langchain_input = {"query": "What did the president say about Ketanji Brown Jackson"}
+    with _mock_request(return_value=_mock_chat_completion_response()):
+        result = loaded_pyfunc_model.predict([langchain_input])
+        assert result == [TEST_CONTENT]
+
+    inference_payload = json.dumps({"inputs": langchain_input})
+    langchain_output_serving = {"predictions": _chat_completion_json_sample(TEST_CONTENT)}
+    with _mock_request(return_value=_MockResponse(200, langchain_output_serving)):
```

Comment:
Is this mock called?

Reply:
I followed the code here:
https://github.com/mlflow/mlflow/blob/3bf587be678630a340a7752f3f32f0f7cb293eee/tests/langchain/test_langchain_model_export.py#L257-L282

Are there any problems with using the same pattern here?

Reply:
`pyfunc_serve_and_score_model` serves a model in a subprocess. Mock only affects the current process.

Reply:
Sure. Added the assertion.